src/algebra/char_p/exp_char.lean

```diff
@@ -0,0 +1,134 @@
+/-
+Copyright (c) 2021 Jakob Scholbach. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jakob Scholbach
+-/
+import algebra.char_p.basic
+import algebra.char_zero
+import data.nat.prime
+
+/-!
+# Exponential characteristic
+
+This file defines the exponential characteristic and establishes a few basic results relating
+it to the (ordinary characteristic).
+The definition is stated for a semiring, but the actual results are for nontrivial rings
+(as far as exponential characteristic one is concerned), respectively a ring without zero-divisors
+(for prime characteristic).
+
+## Main results
+- `exp_char`: the definition of exponential characteristic
+- `exp_char_is_prime_or_one`: the exponential characteristic is a prime or one
+- `char_eq_exp_char_iff`: the characteristic equals the exponential characteristic iff the
+  characteristic is prime
+
+## Tags
+exponential characteristic, characteristic
+-/
+
+universe u
+variables (R : Type u)
+
+section semiring
+
+variables [semiring R]
+
+/-- The definition of the exponential characteristic of a semiring. -/
+class exp_char (R : Type u) [semiring R] (q : ℕ) : Prop :=
+(exp_char_def : (q = 1 ∧ char_zero R) ∨ (q.prime ∧ char_p R q))
```

Comment:
Did you see my commeny on zulip about making this an inductive type instead of an or? If not, I'll dig it up and repost it here

Reply:
Sorry -- yes I did see it but somehow failed to act on it. Maybe you can still repost it here? I will then take care of it. Thanks.

Reply:
sure, [here's the message](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Non-structure.20.60class.60es/near/230668466), which suggests
```lean
class inductive exp_char (R : Type u) [semiring R] (q : ℕ) : Prop
| zero (hq : q = 1) [char_zero R] : exp_char
| prime (hq : q.prime) [char_p R q] : exp_char
```
If you know of better names than `zero` and `prime` feel free to use them; I have no mathematical background here.

Reply:
Or probably even better:
```
class inductive exp_char (R : Type u) [semiring R] (q : ℕ) : ℕ → Prop
| zero [char_zero R] : exp_char 1
| prime (hq : q.prime) [char_p R q] : exp_char q
```

Reply:
This is now implemented.